packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2807,11 +2807,56 @@ class _OffstageElement extends SingleChildRenderObjectElement {
 /// 16.0/9.0. If the maximum width is infinite, the initial width is determined
 /// by applying the aspect ratio to the maximum height.
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// This example shows when width constraint is infinite, height gets biggest allowed
+/// height and width gets height * aspectRatio.
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Container(
+///     color: Colors.purple,
+///     alignment: Alignment.center,
+///     width: double.infinity,
+///     height: 100.0,
+///     child: AspectRatio(
+///     aspectRatio: 16/9,
+///     child: new Container(
+///     color: Colors.amber,
+///       ),
+///     ),
+///  );
+/// }
+/// ```
+/// {@end-tool}
+///
 /// Now consider a second example, this time with an aspect ratio of 2.0 and
 /// layout constraints that require the width to be between 0.0 and 100.0 and
 /// the height to be between 0.0 and 100.0. We'll select a width of 100.0 (the
 /// biggest allowed) and a height of 50.0 (to match the aspect ratio).
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// This example shows when width constraint is finite, width gets biggest allowed
+/// width and height gets width / aspectRatio.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Container(
+///     color: Colors.purple,
+///     alignment: Alignment.center,
+///     width: 100.0,
+///     height: 100.0,
+///     child: AspectRatio(
+///     aspectRatio: 2.0,
+///     child: new Container(
+///     color: Colors.amber,
+///       ),
+///     ),
+///  );
```

Comment:
I think you may have misrepresented the example here a little bit. Here, you want the outer Container to allow its child to have a height and width to be anywhere between 0 to a 100. However, you want the inner widget to want a fixed width and height that initially satisfies the constraint:
```dart
Container(
  color: Colors.green,
  alignment: Alignment.center,
  // child can have height and width between 0 - 100.
  width: 100.0,
  height: 100.0,
  child: AspectRatio(
    // sets the aspect ratio to 2.0.
    aspectRatio: 2.0,
    // have [AspectRatio]'s child
    // be a fixed size of width = 100
    // and height = 50.
    child: Container(
      width: 100,
      height: 50,
      color: Colors.blue,
    ),
  ),
);
```

<img width="97" alt="Screen Shot 2020-11-01 at 1 20 35 PM" src="https://user-images.githubusercontent.com/27032613/97795752-08269700-1c45-11eb-9612-4940b789a257.png">

Then later, they want you to update the AspectRatio, but keep the size constraints intact to see what happens:
```dart
Container(
  color: Colors.green,
  alignment: Alignment.center,
  // child can have height and width between 0 - 100.
  width: 100.0,
  height: 100.0,
  child: AspectRatio(
    // now, try an aspectRatio of 0.5.
    aspectRatio: 0.5,
    // have [AspectRatio]'s child
    // be a fixed size of width = 100
    // and height = 50.
    child: Container(
      width: 100,
      height: 50,
      color: Colors.blue,
    ),
  ),
);
```
This will give you an outcome like this:

<img width="97" alt="Screen Shot 2020-11-01 at 1 19 27 PM" src="https://user-images.githubusercontent.com/27032613/97795736-df060680-1c44-11eb-88c8-a53fd3c2bb6c.png">

As you can see, the child attempts to shape itself according to the aspect ratio over the provided size for the nested `Container`.

Reply:
Also, same comments as above regarding indentations, the `new` keyword, and the color used in the Containers.